### PR TITLE
use stopwords in document featurization

### PR DESCRIPTION
**Pull request by yingjieMiao**
@stkem

### CODE REVIEW

**Comment by eishay** [comment]
can't merge?

File: kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala [hunk 1]
```diff
@@ -22,7 +23,9 @@ abstract class NaiveSumDocRepresenter[M <: StatModel](
   private def wordCounts(doc: Document): Map[String, Int] = {
     val m = mutable.Map[String, Int]()
```
Comment (eishay) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala]: would be nice to have variable names with more then one character (unless its a very small scope as single line per @2is10 rule of thumb)

File: kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala [hunk 2]
```diff
@@ -22,7 +23,9 @@ abstract class NaiveSumDocRepresenter[M <: StatModel](
...
     doc.tokens.foreach { t =>
-      m(t) = m.getOrElse(t, 0) + 1
+      if (stopwords.isEmpty || !stopwords.get.contains(t)) {
+        m(t) = m.getOrElse(t, 0) + 1
```
Comment (eishay) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala]: would creating the map as

``` scala
val m = mutable.Map[String, Int]().withDefaultValue(0)
```

help in this case ?

File: kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala (hunk 1 above)
Comment (yingjieMiao) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala]: it's used within 5 lines :-)  but I can change that.

File: kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala (hunk 2 above)
Comment (yingjieMiao) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala]: yes!

File: .gitignore [hunk 3]
```diff
@@ -3,4 +3,5 @@ workspace.xml
 **/Thumbs.db
 **/*~
 
+target/*
```
Comment (stkem) [inline .gitignore:6]: hmm, how did that happen?

File: kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala [hunk 4]
```diff
@@ -20,11 +21,13 @@ abstract class NaiveSumDocRepresenter[M <: StatModel](
   protected def normalize(vec: Array[Float]): Array[Float]
 
   private def wordCounts(doc: Document): Map[String, Int] = {
-    val m = mutable.Map[String, Int]()
```
Comment (stkem) [inline kifi-backend/modules/cortex/app/com/keepit/cortex/features/DocumentFeature.scala:23]: also in favor of a descriptive name for `m`

**Comment by stkem** [comment]
small comment on naming, lgtm otherwise

**Pull request merged**